nematus/model_updater.py

```diff
@@ -272,8 +319,9 @@ def trim_arrays(arrays, new_seq_lens):
         # source and target tokens. Note that this counts actual tokens
         # (up to and including the <EOS> tokens) not the capacity of the
         # sub-batch.
-        weights = [numpy.sum(s) + numpy.sum(t)
-                      for s, t in zip(split_x_mask, split_y_mask)]
+        # TODO: loss is calculated according to target side, hence here should be weighted by target tokens only.
```

Comment:
Not sure whether this is a mistake. Change the calculation of weights by excluding source tokens.